Intersect.Server/Entities/Entity.cs

```diff
@@ -381,189 +381,150 @@ public virtual void Update(long timeMs)
 
-                        switch (((MapSlideAttribute)tileAttribute).Direction)
-                        {
-                            case 1:
-                                if (moveDir == Direction.Down)
-                                {
-                                    return -4;
-                                }
+        protected virtual bool GetTileBlocker(MapController mapController, int tileX, int tileY, int z,
+            out MovementBlockerType blockerType,
+            out EntityType entityType)
+        {
+            if (mapController == null)
+            {
+                blockerType = MovementBlockerType.OutOfBounds;
+                entityType = EntityType.GlobalEntity;
+                return false;
+            }
 
-                                break;
-                            case 2:
-                                if (moveDir == Direction.Up)
-                                {
-                                    return -4;
-                                }
+            var tileAttribute = mapController.Attributes[tileX, tileY];
 
-                                break;
-                            case 3:
-                                if (moveDir == Direction.Right)
-                                {
-                                    return -4;
-                                }
+            switch (tileAttribute)
+            {
+                case MapAnimationAttribute animAttr when animAttr.IsBlock:
+                case MapBlockedAttribute _:
+                case MapNpcAvoidAttribute _
+                    when (this is Npc || (this is EventPageInstance evtPage && !evtPage.MyPage.IgnoreNpcAvoids)):
+                    blockerType = MovementBlockerType.MapAttribute;
+                    entityType = EntityType.GlobalEntity;
+                    return false;
 
-                                break;
-                            case 4:
-                                if (moveDir == Direction.Left)
-                                {
-                                    return -4;
-                                }
+                case MapZDimensionAttribute zAttr when zAttr.BlockedLevel > 0 && zAttr.BlockedLevel - 1 == Z:
+                    blockerType = MovementBlockerType.ZDimension;
+                    entityType = EntityType.GlobalEntity;
+                    return false;
 
-                                break;
-                        }
-                    }
-                }
-            }
-            else
-            {
-                return -5; //Out of Bounds
+                case MapSlideAttribute _:
+                    blockerType = MovementBlockerType.Slide;
+                    entityType = EntityType.GlobalEntity;
+                    return false;
             }
 
             if (!Passable)
             {
-                var targetMap = mapController;
                 var mapEntities = new List<Entity>();
+                var zoneType = (int)mapController.ZoneType;
```

Comment:
After this line put `var blockableByPlayer = !(this is Player) || !Options.Instance.Passability.Passable[zoneType]`